rust/libnewsboat/src/configcontainer.rs

```diff
@@ -0,0 +1,1360 @@
+                return Err(ConfigHandlerError::TooManyParams);
+            }
+
+            let value = if entry.multi_option {
+                params.join(" ")
+            } else {
+                params[0].clone()
+            };
+
+            match entry.set_value(value) {
+                Ok(_) => Ok(()),
+                Err(msg) => Err(ConfigHandlerError::InvalidParams(msg)),
+            }
+        } else {
+            Err(ConfigHandlerError::InvalidCommand)
+        }
+    }
+
+    pub fn get_configvalue_as_int(&self, key: &str) -> i32 {
+        self.get_configvalue(key).parse().unwrap_or(0)
+    }
+
+    pub fn get_configvalue_as_bool(&self, key: &str) -> bool {
+        let val = self.get_configvalue(key);
+        val == "true" || val == "yes"
+    }
+
+    pub fn get_configvalue_as_filepath(&self, key: &str, path: &mut std::path::PathBuf) -> bool {
+        let val = self.get_configvalue(key);
+        if !val.is_empty() {
+            *path = std::path::PathBuf::from(val);
+            return true;
+        }
+        false
+    }
+
+    pub fn get_configvalue(&self, key: &str) -> String {
+        let data = self.config_data.lock().unwrap();
+        match data.get(key) {
+            Some(entry) => entry.value.clone(),
+            None => String::new(),
+        }
+    }
+
+    pub fn set_configvalue(&self, key: &str, value: &str) -> Result<(), String> {
+        let mut data = self.config_data.lock().unwrap();
+        match data.get_mut(key) {
+            Some(entry) => entry.set_value(value.to_string()),
+            None => Err(gettext("unknown config option: %s").replace("%s", key)),
+        }
+    }
+
+    pub fn reset_to_default(&self, key: &str) {
+        let mut data = self.config_data.lock().unwrap();
+        if let Some(entry) = data.get_mut(key) {
+            entry.value = entry.default_value.clone();
+        }
+    }
+
+    pub fn toggle(&self, key: &str) {
+        let mut data = self.config_data.lock().unwrap();
+        if let Some(entry) = data.get_mut(key)
+            && let ConfigDataType::Bool = entry.data_type
+        {
+            let current = entry.value == "true" || entry.value == "yes";
+            entry.value = if current {
+                "false".to_string()
+            } else {
+                "true".to_string()
+            };
+        }
+    }
+
+    pub fn dump_config(&self) -> Vec<String> {
+        let data = self.config_data.lock().unwrap();
+        let mut output = Vec::new();
+
+        for (key, entry) in data.iter() {
+            let formatted_value = match entry.data_type {
+                ConfigDataType::Bool | ConfigDataType::Int => entry.value.clone(),
+                _ => format!("\"{}\"", entry.value.replace("\"", "\\\"")),
+            };
+
+            let mut line = format!("{key} {formatted_value}");
+            if entry.value != entry.default_value {
+                line.push_str(&format!(" # default: {}", entry.default_value));
+            }
```

Comment:
For multi-options, this puts the entire value in quotes instead of quoting each value individually. A test for this would be appreciated :)

Also, please use `utils::quote` instead of re-implementing it.